app/models/state_changers.rb

```diff
@@ -58,6 +58,23 @@ def self.lookup_for(purpose_uuid)
     details[:state_changer_class].constantize
   end
 
+  # The tube rack state changer is used by TubeRacks.
+  # It contains tubes.
+  class TubeRackStateChanger < DefaultStateChanger
+    # TubeRacks contain tubes
+    def contents_for(_target_state)
+      return nil unless FILTER_FAILS_ON.include?(target_state)
+
+      # determine list of tubes requiring the state change
+      tubes_locations_filtered = labware.tubes.reject { |t| t.state == 'failed' }.map(&:racked_tube.coordinate)
```

Comment:
Discussed whether we should be filtering out 'cancelled' tubes here too? Have this - https://github.com/sanger/limber/pull/2034/files

Also need to test failing and cancelling at the tube rack level. 

Think about whether to create a new story for any of these things.